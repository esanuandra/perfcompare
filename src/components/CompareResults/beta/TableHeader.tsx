import TableCell from '@mui/material/TableCell';
import TableHead from '@mui/material/TableHead';
import TableRow from '@mui/material/TableRow';
import { style } from 'typestyle';

import { Colors, Spacing } from '../../../styles';

const styles = {
  tableHeader: style({
    alignItems: 'center',
    background: Colors.Background100,
    borderRadius: '4px',
    display: 'flex',
    flexDirection: 'row',
    maxWidth: '950px',
    padding: Spacing.Small,
  }),
  filter: style({
    background: Colors.Background200,
    borderRadius: '4px',
  }),
};
function TableHeader() {
  const headerCells = [
    {
      name: 'Platform',
      disable: true,
      filter: true,
      sort: true,
    },
    {
      name: 'Base',
    },
    { name: 'New' },
    {
      name: 'Status',
      disable: true,
      filter: true,
      sort: true,
    },
    {
      name: 'Delta(%)',
    },
    {
      name: 'Confidence',
      disable: true,
      filter: true,
      sort: true,
    },
    { name: 'Total Runs' },
  ];
  return (
    <TableHead className={styles.tableHeader} data-testid='table-header'>
      <TableRow>
        {headerCells.map((header) => (
          <TableCell key={header.name + 'header'} className={header.filter ? 'filter' : ''}>{header.name}</TableCell>
        ))}
      </TableRow>
    </TableHead>
  );
}

export default TableHeader;
