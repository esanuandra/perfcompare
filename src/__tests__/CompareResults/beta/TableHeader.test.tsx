import TableHeader from '../../../components/CompareResults/beta/TableHeader';
import { renderWithRouter } from '../../utils/setupTests';
import { screen } from '../../utils/test-utils';

describe('Table Header', () => {
  it('Should match snapshot', () => {
    renderWithRouter(<TableHeader />);

    expect(screen.getByTestId('table-header')).toBeInTheDocument();
    expect(document.body).toMatchSnapshot();
  });
});