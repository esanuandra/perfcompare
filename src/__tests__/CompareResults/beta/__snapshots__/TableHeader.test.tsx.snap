// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Header Should match snapshot 1`] = `
<body>
  <div>
    <thead
      class="MuiTableHead-root f1wbvtga css-15wwp11-MuiTableHead-root"
      data-testid="table-header"
    >
      <tr
        class="MuiTableRow-root MuiTableRow-head css-1yt2i5z-MuiTableRow-root"
      >
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium filter css-hwty3m-MuiTableCell-root"
          scope="col"
        >
          Platform
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-hwty3m-MuiTableCell-root"
          scope="col"
        >
          Base
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-hwty3m-MuiTableCell-root"
          scope="col"
        >
          New
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium filter css-hwty3m-MuiTableCell-root"
          scope="col"
        >
          Status
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-hwty3m-MuiTableCell-root"
          scope="col"
        >
          Delta(%)
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium filter css-hwty3m-MuiTableCell-root"
          scope="col"
        >
          Confidence
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-hwty3m-MuiTableCell-root"
          scope="col"
        >
          Total Runs
        </th>
      </tr>
    </thead>
  </div>
</body>
`;
